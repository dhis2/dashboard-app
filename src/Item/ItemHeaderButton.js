import React from 'react';
import SvgIcon from 'd2-ui/lib/svg-icon/SvgIcon';

import './ItemHeaderButton.css';

const ItemHeaderButton = ({ icon, onClick, style = {} }) => {
    return (
        <button
            className="item-action-button"
            style={style.container}
            onClick={onClick}
        >
            <SvgIcon icon={icon} style={style.icon} />
        </button>
    );
};

export default ItemHeaderButton;
