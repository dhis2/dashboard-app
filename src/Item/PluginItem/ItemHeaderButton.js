import React from 'react';

const ItemHeaderButton = ({ text, onButtonClick, style }) => (
    <button type="button" onClick={onButtonClick} style={style}>
        {text}
    </button>
);

export default ItemHeaderButton;
