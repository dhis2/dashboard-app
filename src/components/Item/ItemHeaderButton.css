.item-action-button {
    border: none;
    background: transparent;
    cursor: pointer;
}

.item-action-button:focus {
    outline: 0;
}
