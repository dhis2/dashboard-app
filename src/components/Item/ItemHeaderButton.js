import React from 'react';

import './ItemHeaderButton.css';

const ItemHeaderButton = ({ disabled, onClick, children, style }) => {
    return (
        <button
            disabled={disabled}
            type="button"
            className="item-action-button"
            style={style}
            onClick={onClick}
        >
            {children}
        </button>
    );
};

export default ItemHeaderButton;
