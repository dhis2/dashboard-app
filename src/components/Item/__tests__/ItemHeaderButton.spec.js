import React from 'react';
import { shallow } from 'enzyme';
import ItemHeaderButton from '../ItemHeaderButton';

it('renders correctly', () => {
    const button = shallow(
        <ItemHeaderButton
            onClick={Function.prototype}
            style={{ fill: 'purple' }}
        >
            My Little Pony
        </ItemHeaderButton>
    );
    expect(button).toMatchSnapshot();
});
