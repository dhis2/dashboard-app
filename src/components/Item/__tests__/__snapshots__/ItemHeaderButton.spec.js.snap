// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemHeaderButton
    onClick={[Function]}
    style={
      Object {
        "fill": "purple",
      }
    }
  >
    My Little Pony
  </ItemHeaderButton>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "My Little Pony",
      "className": "item-action-button",
      "disabled": undefined,
      "onClick": [Function],
      "style": Object {
        "fill": "purple",
      },
      "type": "button",
    },
    "ref": null,
    "rendered": "My Little Pony",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "My Little Pony",
        "className": "item-action-button",
        "disabled": undefined,
        "onClick": [Function],
        "style": Object {
          "fill": "purple",
        },
        "type": "button",
      },
      "ref": null,
      "rendered": "My Little Pony",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
